tools/scenario-player/scenario_player/tasks/blockchain.py

```diff
@@ -103,12 +105,27 @@ def __init__(
                 'Not all required keys provided. Required: ' + ', '.join(required_keys),
             )
 
-        self.contract_name = config.get('contract_name', None)
-        self.event_name = config.get('event_name', None)
-        self.num_events = config.get('num_events', 0)
+        self.contract_name = config['contract_name']
+        self.event_name = config['event_name']
+        self.num_events = config['num_events']
 
         self.web3 = self._runner.client.web3
 
+        # get the correct contract address
+        service_contract_data = get_contracts_deployed(
+            chain_id=self._runner.chain_id,
+            version=DEVELOPMENT_CONTRACT_VERSION,
+            services=True,
+        )
+        if self.contract_name == CONTRACT_TOKEN_NETWORK:
+            self.contract_address = self._runner.token_network_address
+        else:
+            try:
+                contract_info = service_contract_data['contracts'][self.contract_name]
+                self.contract_address = contract_info['address']
```

Comment:
Looks like `self.contract_address` isn't used anywhere. I would guess it's supposed to be in L133?

Reply:
Nice find 👀